std/assembly/array.ts

```diff
@@ -356,6 +359,66 @@ export class Array<T> {
     }
   }
 
+  join(separator: string = ","): string {
+    var lastIndex = this.length_ - 1;
+    if (lastIndex < 0) return "";
+    var result = "";
+    var value: T;
+    var buffer = this.buffer_;
+    var hasSeparator = separator.length != 0;
+    if (value instanceof bool) {
+      for (let i = 0; i < lastIndex; ++i) {
+        result += loadUnsafe<T,T>(buffer, i) ? "true" : "false";
```

Comment:
Without a GC, string concatenation leaves behind unreferenced instances and building the string from scratch using low-level ops might be a better strategy there. Ideally, just the final step, when appending lastIndex, makes a proper string that can be handled by GC.

Reply:
For `itoa` and especially `dtoa` this pretty problematic because I don't forecast actual string length produced after stringify and can't preallocate one buffer. But I could allocate one buffer with maximum possible size. WDYT?

Reply:
Also this require two pass (interation through array) logic